R/knitr_engine.R

```diff
@@ -38,16 +38,23 @@ eng_impl <- function(options, rextendr_fun) {
   code_out <- glue_collapse(options$code, sep = "\n") # code to output to html
 
   # engine.opts is a list of arguments to be passed to rust_eval, e.g.
-  # engine.opts = list(dependencies = 'pulldown-cmark = "0.8"')
+  # engine.opts = list(dependencies = list(`pulldown-cmark` = "0.8"))
   opts <- options$engine.opts
 
-  if (!is.environment(opts$env)) opts$env <- knitr::knit_global() # default env is knit_global()
+  if (!is.environment(opts$env)) {
+    # default env is knit_global()
+    opts$env <- knitr::knit_global()
+  }
+
+  ui_v("Compiling Rust extendr code chunk...")
+  compilation <- do.call(extendr_engine, c(list(code = code), opts))
```

Comment:
Maybe call this `compiled` or `compiled_code` or `compiled_rust` instead of `compilation`? To me, `compilation` refers to the process of compiling, not the result of it.